app/views/RoomsListView/index.js

```diff
@@ -189,6 +190,10 @@ class RoomsListView extends React.Component {
 				this.backHandler.remove();
 			}
 		});
+		const hasPermissions = ['create-c', 'create-d', 'create-p', 'view-c-room', 'view-d-room', 'view-l-room'];
+		const permissions = await RocketChat.hasPermissionsByUserRoles(hasPermissions);
+		this.setState({ permissions });
+		this.getSubscriptions(true);
```

Comment:
just add a new condition on `componentDidUpdate` to check if `permissions` changed on [this conditional](https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/develop/app/views/RoomsListView/index.js#L285) instead of call `this.getSubscriptions` here.